src/components/SQForm/SQFormMaskedTextField.js

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import MaskedInput from 'react-text-mask';
+import SQFormTextField from './SQFormTextField';
+
+function TextFieldMask({inputRef, mask, ...rest}) {
+  return (
+    <MaskedInput
+      {...rest}
+      ref={ref => {
+        inputRef(ref ? ref.inputElement : null);
+      }}
+      mask={mask}
+      placeholderChar={'\u2000'}
+      showMask={false}
+    />
+  );
+}
+
+function SQFormMaskedTextField({
+  mask,
+  name,
+  label,
+  isDisabled,
+  isRequired,
+  placeholder,
+  size,
+  onBlur,
+  onChange,
+  startAdornment,
+  endAdornment,
+  type,
+  InputProps,
+  inputProps,
+  muiFieldProps
+}) {
+  return (
+    <SQFormTextField
+      name={name}
+      label={label}
+      isDisabled={isDisabled}
+      isRequired={isRequired}
+      placeholder={placeholder}
+      size={size}
+      onBlur={onBlur}
+      onChange={onChange}
+      startAdornment={startAdornment}
+      endAdornment={endAdornment}
+      type={type}
+      InputProps={{
+        ...InputProps,
+        inputComponent: TextFieldMask
+      }}
+      inputProps={{
+        ...inputProps,
+        mask
+      }}
+      muiFieldProps={muiFieldProps}
+    />
+  );
+}
+
+SQFormMaskedTextField.propTypes = {
```

Comment:
Our proptypes don't match the props we defined in the component definition.

Reply:
Are you referring to the props that get passed to SQFormTextField?

Reply:
Right

Reply:
Gotcha, I'll add that. Still getting used to how propTypes is set up, I'm more familiar with typing props in TypeScript